src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1019,6 +1039,33 @@ int sd_dhcp_server_start(sd_dhcp_server *server) {
                 return r;
         }
 
+
+        if (!server->bind_to_interface) {
+                r = dhcp_network_bind_udp_socket(server->ifindex, INADDR_BROADCAST, DHCP_PORT_SERVER, -1);
+
+                if (r < 0) {
+                        sd_dhcp_server_stop(server);
+                        return r;
+                }
+                server->fd_broadcast = r;
+
+                r = sd_event_add_io(server->event, &server->receive_broadcast,
+                                    server->fd_broadcast, EPOLLIN,
+                                    server_receive_message, server);
+                if (r < 0) {
+                        sd_dhcp_server_stop(server);
+                        return r;
+                }
+
+                r = sd_event_source_set_priority(server->receive_broadcast,
+                                                 server->event_priority);
+                if (r < 0) {
+                        sd_dhcp_server_stop(server);
+                        return r;
```

Comment:
one of those days we should replace this error path by a goto label at the end of this function that calls `sd_dhcp_server_stop()` and returns r, we do this so often already in this function

Reply:
Not addressed yet.